app/controllers/v1/billings_controller.rb

```
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module V1
+  # This object is in charge of creating a customer and setting their payment plans
+  # by communicating with Stripe through the Stripe object.
+  class BillingsController < ApplicationController
+    before_action :owner?
+
+    def create
+      customer = Stripe::Customer.create(
+        source: billing_params[:token],
+        email: billing_params[:email]
+      )
+
+      current_user.account.update_attributes(customer_id: customer.id)
+
+      subscription = StripeSubscriptionService.new(
+        current_user.account
+      ).subscribe_to(billing_params[:product])
+
+      if subscription
+        render json: {
+          customer: customer.to_json,
+          subscription: subscription.to_json,
+          message: 'Succesfully subscribed'
+        }, status: 200
+      else
+        render json: { error: 'Something went wrong' }, status: 500
+      end
+    end
+
+    private
+
+    def billing_params
+      params.require(:billing)
+            .permit(:email, :token, :product)
+    end
+
+    def owner?
+      return if current_user.account.owner == current_user
```

Should return true || false as per `?` convention

---

But maybe renaming this method is even better `validate_current_user_is_account_owner`

---

insinde the method we can then use `owner?` as defined in the user model

---

I think I want to throw an actual error, so I think the rename is a good suggestion then I can bypass the convention.